src/server/images/models.js

```diff
@@ -23,4 +25,36 @@ export default class ImageManager {
       cb(new Error('database error'));
     });
   }
+  static getImageUrls({itemKey, isThumbnail = false}, cb) {
+    // get all image Urls of item
+    const keys = [];
+    const promises = [];
+    const checkState = [STATE.ALIVE, STATE.EXPIRED];
+    for (const state of checkState) {
+      promises.push(new Promise((resolve, reject) => {
+        const prefix = KeyUtils.getPrefix(ENTITY.IMAGE, state, itemKey);
+        fetchPrefix(prefix, (err, data) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+          data.map((value) => {
+            keys.push(value.key);
+          });
+          resolve();
+        });
+      }));
+    }
+    Promise.all(promises)
+    .then(() => {
+      const s3Connector = new S3Connector();
+      const urls = (isThumbnail) ?
```

Comment:
right. after then I think making other function for this then() block would be good.